### ISSUE TITLE
Bug: Invalid link generation when using [Link to Text Fragment] extension

### ISSUE THREAD

**Issue by gaiking-uk** [GoogleChromeLabs/text-fragments-polyfill#150]
### Issue

The [Link to Text Fragment] extension does not seem to generate the correct text fragment links if the selected text contains "-"

### Example

- Page: https://imagemagick.org/script/download.php
- Selected text: Portable Win64 static at 16 bits-per-pixel component with
- ✅ Generated link (using Built-in): https://imagemagick.org/script/download.php#:~:text=x64%2Dstatic.exe-,Win64%20static%20at%2016%20bits%2Dper%2Dpixel%20component%20with,-high%20dynamic%2Drange
- ❌ Generated link (using Extension):
https://imagemagick.org/script/download.php#:~:text=x64-static.exe-,win64%20static%20at%2016%20bits-per-pixel%20component%20with,-high%20dynamic-range

![TextFragementLink Screenshot](https://user-images.githubusercontent.com/37931190/231166781-c47cd24e-02a7-4d39-ae2a-d34a565d0916.jpg)

### System / Technical Info
- O/S: Windows 11
- Browser: Edge
- Browser version: 112.0.1722.34 (Official build) (64-bit)
- Extension version: 2.3.2

**Comment by tomayac** [comment]
Indeed manually percent-encoding the `-` to `%2D` in `prefix`, `start`, and `suffix` makes the extension-generated link work: [`https://imagemagick.org/script/download.php#:~:text=x64%2Dstatic.exe-,win64%20static%20at%2016%20bits%2Dper%2Dpixel%20component%20with,-high%20dynamic%2Drange`](https://imagemagick.org/script/download.php#:~:text=x64%2Dstatic.exe-,win64%20static%20at%2016%20bits%2Dper%2Dpixel%20component%20with,-high%20dynamic%2Drange).

**Comment by tomayac** [comment]
Potentially connected to https://github.com/GoogleChromeLabs/text-fragments-polyfill/issues/115, which likewise is about percent-encoding.

**Comment by tomayac** [comment]
The requirement for percent-encoding `-` is specifically [mentioned in the spec](https://wicg.github.io/scroll-to-text-fragment/#:~:text=If%20a%20text%20fragment%20refers%20to%20a%20%22%26%22%2C%20%22%2D%22%2C%20or%20%22%2C%22%20character%20in%20the%20document%2C%20it%20will%20be%20percent%2Dencoded%20in%20the%20fragment.).

**Comment by gaiking-uk** [comment]
> Indeed manually percent-encoding the `-` to `%2D` in `prefix`, `start`, and `suffix` makes the extension-generated link work: [`https://imagemagick.org/script/download.php#:~:text=x64%2Dstatic.exe-,win64%20static%20at%2016%20bits%2Dper%2Dpixel%20component%20with,-high%20dynamic%2Drange`](https://imagemagick.org/script/download.php#:~:text=x64%2Dstatic.exe-,win64%20static%20at%2016%20bits%2Dper%2Dpixel%20component%20with,-high%20dynamic%2Drange).


Yeah, is weird though -- if you encode the full string (which contains ***multiple*** "-" characters) that link doesn't seem to work either, AFAICT (can tell) it seems to explicitly require that ***some*** of the "-" are encoded to %2D and some are left as "-"(?) 😐🤔

Anyway, is a bit beyond my level am afraid so will leave it in your capable hands!
-- Thanks again for replying and assisting on this so promptly! 👍🏼

**Comment by tomayac** [comment]
The ones that don’t need encoding are the separators of the prefix and the suffix. See https://web.dev/text-fragments/ for a (hopefully) accessible explanation.
